cpp_tests/feature/one_hot_encoder_test.cc

```diff
@@ -81,9 +82,53 @@ TEST(OneHotEncoderTest, Load) {
   auto model = clink::OneHotEncoderModel::load(dir_name);
   EXPECT_EQ(!model.takeError(), true);
 
-  auto vector = model->transform(1, 0);
+  auto vector = (*model)->transform(1, 0);
   EXPECT_EQ(!vector.takeError(), true);
-  EXPECT_EQ(vector->get(1).get(), 1.0);
+  EXPECT_EQ((*vector)->get(1).get(), 1.0);
+
+  clink::test::deleteFolderRecursively(dir_name);
+}
+
+TEST(OneHotEncoderTest, Mlir) {
+  std::string dir_name = clink::test::createTemporaryFolder();
+
+  nlohmann::json params;
+  params["paramMap"]["dropLast"] = "false";
+
+  std::ofstream params_output(dir_name + "/metadata");
+  params_output << params;
+  params_output.close();
+
+  clink::OneHotEncoderModelDataProto model_data;
+  model_data.add_featuresizes(2);
+  model_data.add_featuresizes(3);
+
+  mkdir((dir_name + "/data").c_str(), S_IRWXU);
+  std::ofstream model_data_output(dir_name + "/data/" +
+                                  clink::test::generateRandomString());
+  model_data.SerializeToOstream(&model_data_output);
+  model_data_output.close();
+
+  auto mlir_script = R"mlir(
+    func @main(%path: !tfrt.string, %value: i32, %column_index: i32) -> !tfrt.string {
+      %ch0 = tfrt.new.chain
+      %model = clink.feature.onehotencoder.load %path
+      %vector = clink.feature.onehotencoder.transform %model, %value, %column_index
+      %vector_string = clink.linalg.sparsevector.tostring %vector
+      tfrt.return %vector_string : !tfrt.string
+    }
+  )mlir";
+
+  // Executes ClinkRunner.
+  llvm::SmallVector<RCReference<AsyncValue>> inputs;
+  inputs.push_back(tfrt::MakeAvailableAsyncValueRef<std::string>(dir_name));
+  inputs.push_back(tfrt::MakeAvailableAsyncValueRef<int32_t>(1));
+  inputs.push_back(tfrt::MakeAvailableAsyncValueRef<int32_t>(0));
+
+  auto results = clink::test::runMlirScript(mlir_script, inputs);
+
+  EXPECT_EQ(results.size(), 1);
+  EXPECT_EQ(results[0]->get<std::string>(), "(2, (1), (1.000000)");
```

Comment:
Future developers might use existing tests as examples when developing new tests. It could be useful to setup some best practice from the beginning.

In general some kernel might have asynchronous operation and the AsyncValue in `results ` might not be available yet. We probably should wait explicitly wait for results to be ready before calling `get<std::string>()`.